test/functional/uvm_scsi_test.go

```diff
@@ -64,33 +68,88 @@ func testSCSIAddRemove(t *testing.T, u *uvm.UtilityVM, pathPrefix string, operat
 
 	// Add each of the disks to the utility VM. Attach-only, no container path
 	logrus.Debugln("First - adding in attach-only")
+	for i := 0; i < numDisks; i++ {
+		_, _, _, err := u.AddSCSI(context.Background(), disks[i], "", false)
+		if err != nil {
+			t.Fatalf("failed to add scsi disk %d %s: %v", i, disks[i], err)
+		}
+	}
+
+	// Remove them all
+	logrus.Debugln("Removing them all")
+	for i := 0; i < numDisks; i++ {
+		if err := u.RemoveSCSI(context.Background(), disks[i]); err != nil {
+			t.Fatalf("failed to remove SCSI disk: %v", err)
+		}
+	}
+
+	// Now re-add but providing a container path
+	logrus.Debugln("Next - re-adding with a container path")
 	for i := 0; i < numDisks; i++ {
 		_, _, _, err := u.AddSCSI(context.Background(), disks[i], fmt.Sprintf(`%s%d`, pathPrefix, i), false)
 		if err != nil {
-			t.Fatalf("failed to add scsi disk %d %s: %s", i, disks[i], err)
+			t.Fatalf("failed to add scsi disk %d %s: %v", i, disks[i], err)
+		}
+	}
+
+	logrus.Debugln("Next - Removing them")
+	for i := 0; i < numDisks; i++ {
+		if err := u.RemoveSCSI(context.Background(), disks[i]); err != nil {
+			t.Fatalf("failed to remove SCSI disk: %v", err)
+		}
+	}
+
+	// TODO: Could extend to validate can't add a 64th disk (windows). 65th (linux).
+}
+
+func testSCSIAddRemoveMultiple(t *testing.T, u *uvm.UtilityVM, pathPrefix string, operatingSystem string, wcowImageLayerFolders []string) {
+	numDisks := 63 // Windows: 63 as the UVM scratch is at 0:0
+	if operatingSystem == "linux" {
+		numDisks++ //
+	}
+
+	// Create a bunch of directories each containing sandbox.vhdx
+	disks := make([]string, numDisks)
+	for i := 0; i < numDisks; i++ {
+		tempDir := ""
+		if operatingSystem == "windows" {
+			tempDir = testutilities.CreateWCOWBlankRWLayer(t, wcowImageLayerFolders)
+		} else {
+			tempDir = testutilities.CreateLCOWBlankRWLayer(context.Background(), t)
+		}
+		defer os.RemoveAll(tempDir)
+		disks[i] = filepath.Join(tempDir, `sandbox.vhdx`)
+	}
+
+	// Add each of the disks to the utility VM. Attach-only, no container path
+	logrus.Debugln("First - adding in attach-only")
+	for i := 0; i < numDisks; i++ {
+		_, _, _, err := u.AddSCSI(context.Background(), disks[i], "", false)
```

Comment:
We should be able to extract a function `addDisksToUVM(uvm *uvm.UtilityVM, disks []string, pathPrefix string, usePath bool) error` and call it instead of needing 4 loops in this function for each of the adds, I think. Likewise with a `removeDisksFromUVM` function.

Reply:
do you prefer I do that in this PR?

Reply:
I would prefer that since I think it's a fairly quick change. If you think it would take a while we can leave as-is for now.

Reply:
It would make the function a lot more readable.

Reply:
Done.